third_party/xla_client/xrt_computation_client.cc

```diff
@@ -374,33 +380,43 @@ XrtComputationClient::TransferToServerInternal(
 
   mwait->Reset(session_work_map.size());
   std::vector<DataPtr> results(tensors.size());
-  for (auto& session_session_work : session_work_map) {
-    XrtSession* session = session_session_work.first;
-    SessionWork* session_work = &session_session_work.second;
-    auto runner = [&, session, session_work]() {
-      std::vector<tensorflow::Tensor> outputs;
-      XLA_CHECK_OK(session->session()->Run(
-          session_work->feed_inputs, session_work->outputs_handles, &outputs));
-      XLA_CHECK_EQ(outputs.size(), session_work->outputs_handles.size());
-
-      for (size_t i = 0; i < outputs.size(); ++i) {
-        size_t li = session_work->index_mapping[i];
-        results[li] = std::make_shared<XrtData>(this, tensors[li].device,
-                                                tensors[li].shape,
-                                                outputs[i].scalar<int64>()());
-      }
-      CreateDataHandlesCounter()->AddValue(outputs.size());
-    };
-    env::ScheduleIoClosure(
-        util::MultiWait::Completer(mwait, std::move(runner)));
+  {
```

Comment:
What's the purpose of putting the below into a separate block?

Reply:
The `TraceMe` annotations are [RAII](https://en.cppreference.com/w/cpp/language/raii) so I wanted to more clearly spell out that this the `TransferToServerExecute` instance of `TraceMe` produces a trace for this block of code.